lib/ui/home/home_gallery_widget.dart

```diff
@@ -34,8 +35,20 @@ class HomeGalleryWidget extends StatelessWidget {
         final ownerID = Configuration.instance.getUserID();
         final hasSelectedAllForBackup =
             Configuration.instance.hasSelectedAllFoldersForBackup();
-        final collectionsToHide =
+        final archivedOrHiddenCollectionIDs =
             CollectionsService.instance.archivedOrHiddenCollectionIds();
+        final Set<int> collectionsToHide;
+        if (LocalSettings.instance.hideSharedItems) {
+          final incomingSharedCollectionIDs = CollectionsService.instance
+              .getSharedCollections()
+              .incoming
+              .map((e) => e.id)
+              .toSet();
+          collectionsToHide =
+              archivedOrHiddenCollectionIDs.union(incomingSharedCollectionIDs);
```

Comment:
We are using `archivedOrHiddenCollectionIds` at bunch of places. We can simplify this change by modifying the same method.

``` dart
Set<int> archivedOrHiddenCollectionIds() {
    final bool archiveIncomingCollections = LocalSettings.instance.hideSharedItems;
    final int ownerID = _config.getUserID()!;
    final Set<int> result = <int>{};
    for (final collection in _collectionIDToCollections.values) {
      if (collection.isHidden() ||
          collection.isArchived() ||
          collection.hasShareeArchived()) {
        result.add(collection.id);
        continue;
      }
      if (archiveIncomingCollections) {
        if (collection.owner?.id != ownerID) {
          result.add(collection.id);
        }
      }
    }
    return result;
  }
```
cc @ashilkn

Reply:
@ua741 Should I Update it?

Reply:
@ua741 Got it